pkg/template/servicebroker/catalog.go

```diff
@@ -112,9 +112,11 @@ func (b *Broker) Catalog() *api.Response {
 		return api.InternalServerError(err)
 	}
 
-	services := make([]*api.Service, len(templates))
-	for i, template := range templates {
-		services[i] = serviceFromTemplate(template)
+	var services []*api.Service
+	for _, template := range templates {
```

Comment:
can't we just pass the valid namespaces to List()?  I'm concerned about us getting back way more content than we care about and having to filter it here.